bootstrap/config/config.go

```diff
@@ -402,24 +400,28 @@ func (cp *Processor) getAccessTokenCallback(serviceKey string, secretProvider in
 // LoadCustomConfigSection loads the specified custom configuration section from file or Configuration provider.
 // Section will be seed if Configuration provider does yet have it. This is used for structures custom configuration
 // in App and Device services
-func (cp *Processor) LoadCustomConfigSection(config interfaces.UpdatableConfig, sectionName string) error {
-	var overrideCount = -1
-	var err error
-	source := "file"
-
+func (cp *Processor) LoadCustomConfigSection(updatableConfig interfaces.UpdatableConfig, sectionName string) error {
 	if cp.envVars == nil {
 		cp.envVars = environment.NewVariables(cp.lc)
 	}
 
 	configClient := container.ConfigClientFrom(cp.dic.Get)
 	if configClient == nil {
 		cp.lc.Info("Skipping use of Configuration Provider for custom configuration: Provider not available")
-		tomlTree, err := cp.loadPrivateFromFile()
+		filePath := GetConfigFileLocation(cp.lc, cp.flags)
+		configMap, err := cp.loadConfigYamlFromFile(filePath)
 		if err != nil {
 			return err
 		}
-		if err := tomlTree.Unmarshal(config); err != nil {
-			return fmt.Errorf("could not load toml tree into custom config: %s", err.Error())
+
```

Comment:
can we use the convertMapToInterface function here?

Reply:
For which part? 
i.e. this comment is for blank line 416.

Reply:
ahh, ok. Thinking...

Reply:
Yep, thanks!

Reply:
reworked